server/datastore/mysql/scripts.go

```diff
@@ -259,8 +299,32 @@ VALUES
 		}
 		return nil, ctxerr.Wrap(ctx, err, "insert script")
 	}
-	id, _ := res.LastInsertId()
-	return ds.getScriptDB(ctx, ds.writer(ctx), uint(id))
+	return res, nil
+}
+
+func insertScriptContents(ctx context.Context, contents string, tx sqlx.ExtContext) (sql.Result, error) {
+	const insertStmt = `
+INSERT INTO
+  script_contents (
+	  md5_checksum, contents
+  )
+VALUES (UNHEX(?),?)
+ON DUPLICATE KEY UPDATE
+  id=LAST_INSERT_ID(id)
```

Comment:
I don't know for sure, but it looks like this could change the actual `id` of the row that existed? I think it should just be `id=id` or I think even safer would be to update a non-key column like `created_at = created_at` just to be super paranoid.

Reply:
Ah, looks like this may actually be a hack to make sure `res.LastInsertID` works and returns the id of the updated row?

Reply:
Yeah, it's a common practice (at least from what I see online) to make sure that the ID is returned regardless of whether or not there was an update. In my manual testing it's worked as expected.